stl/inc/algorithm

```diff
@@ -2900,6 +2900,121 @@ _FwdIt3 transform(_ExPo&& _Exec, const _FwdIt1 _First1, const _FwdIt1 _Last1, co
     _Fn _Func) noexcept; // terminates
 #endif // _HAS_CXX17
 
+#ifdef __cpp_lib_concepts
+namespace ranges {
+
+    // ALIAS TEMPLATE unary_transform_result
+    template <class _In, class _Out>
+    using unary_transform_result = in_out_result<_In, _Out>;
+
+    // ALIAS TEMPLATE binary_transform_result
+    template <class _In1, class _In2, class _Out>
+    using binary_transform_result = in_in_out_result<_In1, _In2, _Out>;
+
+    // VARIABLE ranges::transform
+    class _Transform_fn : private _Not_quite_object {
+    public:
+        using _Not_quite_object::_Not_quite_object;
+
+        // clang-format off
+        template <input_iterator _It, sentinel_for<_It> _Se, weakly_incrementable _Out,
+            copy_constructible _Fn, class _Pj = identity>
+            requires indirectly_writable<_Out, indirect_result_t<_Fn&, projected<_It, _Pj>>>
+        _NODISCARD constexpr unary_transform_result<_It, _Out> operator()(_It _First, _Se _Last, _Out _Result,

```

Comment:
Because `transform` writes to an output iterator, I believe it shouldn't be `_NODISCARD` (corrections appreciated). Occurs below.

The general principle is that pure observers should be marked `_NODISCARD` (most common). Other scenarios are where ignoring the return value is extremely likely to be a leak (e.g. raw allocation functions) or otherwise incorrect (e.g. `std::remove`, where you'll have garbage elements if you don't use the return value to `erase`). If a function has side effects, and it's possible to use correctly while ignoring the return value, then we don't mark it `_NODISCARD`; that applies to most mutating algorithms.

Reply:
Removed `nodiscard` from the `oeprator()` calls but not the internal helpers

Reply:
Agreed, entirely reasonable to keep for the internal helpers. Thanks!